BaseLib/ConfigTreeNew.h

```diff
@@ -399,17 +578,26 @@ class ConfigTreeNew final
     //! A path printed in error/warning messages.
     std::string _path;
 
-    //! A map key -> (count, type) keeping track which parameters have been read how often
-    //! and which datatype they have.
+    //! The path of the file from which this tree has been read.
+    std::string _filename;
+
+    //! A pair (is attribute, tag/attribute name)
+    using KeyType = std::pair<bool, std::string>;
```

Comment:
Maybe you can clearify: Is the first component of the pair true if the second component is a *attribute* name or is it true if the second component is a *tag* name?